### PR TITLE
ONNX FE Refactor: ngraph -> ov

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - Refactoring of the ONNX FE to reduce the ngraph dependencies and switch to the ones defined in the `ov::` namespace
 - Code cleanup and removal of obsolete files

### CODE REVIEW

File: docs/IE_DG/Extensibility_DG/Custom_ONNX_Ops.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Custom ONNX* Operators {#openvino_docs_IE_DG_Extensibility_DG_Custom_ONNX_Ops}
 
 The ONNX\* importer provides a mechanism to register custom ONNX operators based on predefined or custom nGraph operations.
-The function responsible for registering a new operator is called `ngraph::onnx_import::register_operator` and defined in the `onnx_import/onnx_utils.hpp` file.
+The function responsible for registering a new operator is called `ov::onnx_import::register_operator` and defined in the `onnx_import/onnx_utils.hpp` file.
```
Comment (ilyachur) [inline docs/IE_DG/Extensibility_DG/Custom_ONNX_Ops.md:4]: I think `onnx_import::` is a legacy low level API.
I think we shouldn't add it to new API.
new API has an extension mechanism. please keep old namespace for legacy importer API

File: docs/onnx_custom_op/onnx_custom_op.cpp [hunk 2]
```diff
@@ -15,9 +15,9 @@ void register_custom_relu_operator() {
     // x < 0  => f(x) = x * beta
 
 //! [onnx_custom_op:register_operator]
-    ngraph::onnx_import::register_operator(
-        "CustomRelu", 1, "com.example", [](const ngraph::onnx_import::Node& onnx_node) -> ngraph::OutputVector {
-            namespace opset = ngraph::opset5;
+    ov::onnx_import::register_operator(
```
Comment (ilyachur) [inline docs/onnx_custom_op/onnx_custom_op.cpp:18]: I think we need to demonstrate new Extension API

File: src/frontends/onnx/frontend/include/onnx_import/onnx.hpp [hunk 3]
```diff
@@ -10,11 +10,10 @@
 #include <set>
 #include <string>
 
-#include "ngraph/function.hpp"
 #include "onnx_importer_visibility.hpp"
+#include "openvino/core/model.hpp"
 
-/// \brief              Top level nGraph namespace.
-namespace ngraph {
```
Comment (ilyachur) [inline src/frontends/onnx/frontend/include/onnx_import/onnx.hpp:17]: It is legacy API. We shouldn't change namespace here.

File: src/frontends/onnx/frontend/include/onnx_import/core/node.hpp [hunk 4]
```diff
@@ -7,22 +7,22 @@
...
 #include "onnx_import/onnx_importer_visibility.hpp"
+#include "openvino/core/except.hpp"
+#include "openvino/core/node.hpp"
 
 namespace ONNX_NAMESPACE {
 // forward declaration
 class NodeProto;
 }  // namespace ONNX_NAMESPACE
 
-namespace ngraph {
```
Comment (ilyachur) [inline src/frontends/onnx/frontend/include/onnx_import/core/node.hpp:19]: It is not a backward compatible change.

File: src/frontends/onnx/frontend/include/onnx_import/core/operator_set.hpp [hunk 5]
```diff
@@ -8,17 +8,16 @@
 #include <string>
 #include <unordered_map>
 
-#include "ngraph/node.hpp"
 #include "onnx_import/core/node.hpp"
 
-namespace ngraph {
+namespace ov {
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/include/onnx_import/core/operator_set.hpp:13]: can we keep this stuff in old API only?

Comment (tomdol) [reply src/frontends/onnx/frontend/include/onnx_import/core/operator_set.hpp:13]: Yes, I've reverted those changes after a chat with Ilya C.

**Comment by tomdol** [comment]
As a matter of fact, I've started bringing back the symbols already exported as a part of public API to the ngraph namespace. This created a rather ugly mix of `ov::` and `ngraph::` symbols used in various places. I think this refactor should be postponed until the final removal of the legacy API. We should be able to hide the symbols still needed (stop exporting them) and move them to the `ov` namespace entirely.

**Pull request closed without merging**